e2e/tests/negative/node_drain.robot

```diff
@@ -89,3 +93,142 @@ Force Drain Replica Node While Replica Rebuilding
+
+Test kubectl drain nodes for PVC/PV/LHV is created through Longhorn API
+    [Documentation]    Test kubectl drain nodes for PVC/PV/LHV is created through Longhorn API
+    ...    Given 1 PVC/PV/LHV created through Longhorn API And LHV is not yet attached/replicated.
+    ...    When kubectl drain nodes.
+    ...        NODE=centos-worker-0
+    ...        kubectl cordon ${NODE}
+    ...        kubectl drain --force --ignore-daemonsets --delete-emptydir-data --grace-period=10 ${NODE}
+    ...    Then all node should successfully drain.
+    When Create volume 0 with   dataEngine=${DATA_ENGINE}
+    And Wait for volume 0 detached
+    And Create persistentvolume for volume 0
+    And Create persistentvolumeclaim for volume 0
+    And Force drain all nodes
+
+Stopped replicas on deleted nodes should not be counted as healthy replicas when draining nodes
+    [Documentation]    Stopped replicas on deleted nodes should not be counted as healthy replicas when draining nodes
+    ...    When draining a node, the node will be set as unscheduled and all pods should be evicted.
+    ...    By Longhorn’s default settings, the replica will only be evicted if there is another healthy replica on the running node.
+    ...    Related Issue:
+    ...    - https://github.com/longhorn/longhorn/issues/2237
+    ...
+    ...    Given Longhorn with 2 nodes cluster: Node_1, Node_2
+    ...    And Update the Node Drain Policy setting to block-if-contains-last-replica
+    ...    And Create a 5Gi detached volume with 2 replicas.
+    ...    And Stop Node_1 that contains one of the replicas.
+    ...    When Attempts to drain Node_2 that contains remaining replica.
+    ...        kubectl drain <Node_2 name> --delete-emptydir-data=true --force=true --grace-period=-1 --ignore-daemonsets=true
+    ...    Then The Node_2 becomes cordoned.
+    ...    And All pods on Node_2 are evicted except the replica instance manager pod.
+    ...        kubectl get pods --field-selector spec.nodeName=<Node_2 name> -o wide -n longhorn-system
+    ...    And The message like below keeps appearing.
+    ...        evicting pod longhorn-system/instance-manager-r-xxxxxxxx
+    ...        error when evicting pods/"instance-manager-r-xxxxxxxx" -n "longhorn-system" (will retry after 5s): Cannot evict pod as it would violate the pod's disruption budget.
+    ...    And The last healthy replica exists on the Node_2.
+    [Teardown]    Cleanup test resources include off nodes
+    Given Disable node 0 scheduling
+    And Set setting node-drain-policy to block-if-contains-last-replica
+    And Given Create volume 0 with    size=5Gi    numberOfReplicas=2
+    And Attach volume 0 to node 1
+    And Wait for volume 0 healthy
+    And Write data to volume 0
+    And Detach volume 0 from attached node
+    And Wait for volume 0 detached
+    And Power off node 1
+
+    When Force drain node 2 and wait for 90 second    dataEngine=${DATA_ENGINE}
```

Comment:
Test steps of a robot test case should be self-explanatory, but the following steps are unclear in terms of what they are checking:
```
When Force drain node 2 and wait for 90 second dataEngine=${DATA_ENGINE}
And The drain process not completed
And Check instance-manager pod is running on node 2    dataEngine=${DATA_ENGINE}
```
Could these steps be made more readable to clarify what exactly is being validated?

Reply:
Updated,  removed `dataEngine=${DATA_ENGINE}` in `When Force drain node 2 and wait for 90 second    dataEngine=${DATA_ENGINE}`and use `data_engine = BuiltIn().get_variable_value("${DATA_ENGINE}")` to get data engine setting
```python
def get_instance_manager_on_node(node_name):
    data_engine = BuiltIn().get_variable_value("${DATA_ENGINE}")
    pods = get_all_pods_on_node(node_name)
    for pod in pods:
        labels = pod.metadata.labels
        if labels.get("longhorn.io/data-engine") == data_engine and \
           labels.get("longhorn.io/component") == "instance-manager":
            return pod.metadata.name
    return None
```


[Test result](https://ci.longhorn.io/job/private/job/longhorn-e2e-test/1491/)